src/msg/async/AsyncMessenger.cc

```diff
@@ -167,7 +167,7 @@ int Processor::rebind(const set<int>& avoid_ports)
   ldout(msgr->cct, 10) << __func__ << " new nonce " << nonce << " and inst " << msgr->my_inst << dendl;
 
   ldout(msgr->cct, 10) << __func__ << " will try " << addr << " and avoid ports " << new_avoid << dendl;
-  return bind(addr, new_avoid);
+  return bind(msgr->my_inst.addr, new_avoid);
```

Comment:
@yuyuyu101 
I think it is a compiler bug, that does not correctly uses the new value of addr, but an older value (exactly 100000 less_.
When I use `msgr->my_inst.addr` as value for bind, the problem goes away.
And both values are equivalent from user perspective.
So I see no problem in changing the reference, and it helps me in avoiding thru hoops in getting the compiler fixed. (I've reported the problem, but it is very hard to reporduce in an "simple" example.